packages/flutter_test/lib/src/matchers.dart

```diff
@@ -2695,6 +2695,21 @@ class _MatchesSemanticsData extends Matcher {
     return true;
   }
 
+  bool _checkStringMismatch(Map<dynamic, dynamic> matchState, String prefixText, String actual) {
+    final String actualWithNBSP = actual.replaceAll('\u202f', r'\u202f');
+    return failWithDescription(matchState, '$prefixText was $actualWithNBSP');
+  }
+
+  bool _checkStringAttributeMismatch(
+    Map<dynamic, dynamic> matchState,
+    String prefixText,
+    AttributedString actual,
+  ) {
+    final String actualStringWithNBSP = actual.string.replaceAll('\u202f', r'\u202f');
+    final actualWithNBSP = AttributedString(actualStringWithNBSP, attributes: actual.attributes);
+    return failWithDescription(matchState, '$prefixText was: $actualWithNBSP');
```

Comment:
nit: Rename it with "escaped", a more common way to call this process
```suggestion
    final String actualStringEscaped = actual.string.replaceAll('\u202f', r'\u202f');
    final actualEscaped = AttributedString(actualStringEscaped, attributes: actual.attributes);
    return failWithDescription(matchState, '$prefixText was: $actualEscaped');
```